### PR TITLE
DPE-1098 - Github release related changes

### PR DESCRIPTION
**Pull request by averma-canonical**
Includes changes to

- refactor the existing snap publish pipeline with feedback from Alex. PR created explicitly or on spark update detection. publish on PR approval. 
- add github release feature for client python scripts.

### CODE REVIEW

**Review by deusebio — DISMISSED** [review]
CI/CD looks sane! As mentioned on MM, It could just be nice to make sure the cron job works and a PR is created with update on SPARK_VERSION on a personal repository, just to make sure updates are rightfully in place

**Review by deusebio — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      - id: checkout
+        name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.branch }}
+      - id: build-snap
+        name: Build snap
+        uses: snapcore/action-build@v1
+        with:
+          snapcraft-channel: 7.x/candidate
```
Comment (taurus-forever) [inline .github/workflows/ci.yaml:57]: FJMI, is there some fixes for you in 7.2.10?

File: .github/workflows/on_spark_update_available.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          STATUSCODE=$(curl --silent --head https://downloads.apache.org/spark/${LATEST_STABLE_SPARK_VERSION}/${LATEST_STABLE_SPARK_VERSION}-bin-hadoop3.tgz | head -n 1 | cut -d' ' -f2)
+          if  [[ ${LATEST_AVAILABLE_VERSION} != ${CURRENT_PUBLISHED_VERSION} ]] && [[ ${STATUSCODE} -eq 200 ]]
+            then 
+              echo "::set-output name=decision::1"
+              echo "New Spark version available....."
+            else
+              echo "::set-output name=decision::0"
+              echo "No updates to Spark detected!"
+          fi
+          echo "::set-output name=version::${LATEST_AVAILABLE_VERSION}"
```
Comment (taurus-forever) [inline .github/workflows/on_spark_update_available.yaml:44]: Here and above, JFYI, "::set-output name=..." is deprecated:
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

The code will stop working on May 31 - June 1, 2023.

File: .github/workflows/on_spark_update_available.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          path: ./SPARK_VERSION
+          write-mode: overwrite
+          contents: |
+            ${{ needs.check-spark.outputs.version }}
+
+      - name: create pull request
+        id: cpr
+        uses: peter-evans/create-pull-request@v4
+        with:
+          token: ${{ secrets.SPARK_PKG_PUBLISH }}
```
Comment (taurus-forever) [inline .github/workflows/on_spark_update_available.yaml:70]: It is OK to merge this now to proceed, but we will have to refactor this to stop sharing our secrets with 3rd party actions. BTW, which token used here `averma-canonical`? If true, PR will be created from this user and he cannot review PR :-( 

Two options on the table right now:
 * https://github.com/canonical/zookeeper-rock/blob/3.6/stable/.github/workflows/on_new_version_available.yaml#L38
 * https://github.com/canonical/mysql-k8s-bundle/pull/17 (WIP, should be tested first).

File: .github/workflows/release_github.yaml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+          if [[ "$TAG" != "v$PROJECT_VERSION" ]]; then exit 1; fi
+#      - name: Release Notes
+#        uses: heinrichreimer/github-changelog-generator-action@v2.3
+#        with:
+#          token: ${{ secrets.SPARK_PKG_PUBLISH }}
+#          output: ".github/RELEASE-TEMPLATE.md"
+      - name: Create Github Release
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.SPARK_PKG_PUBLISH }}
```
Comment (taurus-forever) [inline .github/workflows/release_github.yaml:83]: Could be one liner https://cli.github.com/manual/gh_release_create
after signing https://docs.google.com/document/d/1OTV6l8LY0uBe4LehpP-Nd3QWB-2qXdxUJ3DYqka3w04/edit#

**Review by taurus-forever — APPROVED** [review]
LGTM for now, but it will require refactoring soon.
Reasons: set-output and 3rd party actions with secrets.

P.S. Nice work!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (averma-canonical) [reply .github/workflows/ci.yaml:57]: Not expecting any. Can change this in future.

File: .github/workflows/on_spark_update_available.yaml (hunk 2 above)
Comment (averma-canonical) [reply .github/workflows/on_spark_update_available.yaml:44]: Thanks, will raise a refactor work item for this.

File: .github/workflows/on_spark_update_available.yaml (hunk 3 above)
Comment (averma-canonical) [reply .github/workflows/on_spark_update_available.yaml:70]: Can be updated separately.

**Pull request merged**